components/clm/src/main/clmfates_interfaceMod.F90

```diff
@@ -206,9 +209,148 @@ module CLMFatesInterfaceMod
 
    character(len=*), parameter, private :: sourcefile = &
         __FILE__
+
+   public  :: ELMFatesGlobalElements
    
 contains
 
+   
+
+   subroutine ELMFatesGlobalElements()
+
+     ! --------------------------------------------------------------------------------
+     ! This is one of the first calls to fates
+     ! Used for setting dimensions.  This MUST
+     ! be called after NL variables are specified and
+     ! after the FATES parameter file has been read in
+     ! Aside from setting global dimension info, which
+     ! is used in the history file, we also transfer
+     ! over the NL variables to FATES global settings.
+     ! --------------------------------------------------------------------------------  
+     
+     integer                                        :: pass_masterproc
+     integer                                        :: pass_vertsoilc
+     integer                                        :: pass_spitfire     
+     integer                                        :: pass_ed_st3
+     integer                                        :: pass_logging
+     integer                                        :: pass_ed_prescribed_phys
+     integer                                        :: pass_planthydro
+     integer                                        :: pass_inventory_init
+     integer                                        :: pass_is_restart
+     
+     if (use_fates) then
```

Comment:
Why not have the following?
   `if (.not. use_fates) return`

Reply:
I can look through this, but there is a small possibility that we need the last call in the routine even if we don't have fates turned on... but I could double check that.

Reply:
well, according to Ryan from the past: https://github.com/E3SM-Project/E3SM/pull/3471/files/55752fa50e3312cdb39d703828772af056cfbb39#diff-26b8ffc0ec3b66f6b87a776e9ee64416R345
But I don't trust that guy

Reply:
@bishtgautam , github code linking has been weird for me with low bandwidth lately, so to expand upon my cheeky comment: as far as I can tell, I think we do actually need some of the code that comes after the "if" clause, so we shouldn't return early.

Reply:
Ok, sounds good.